src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -136,6 +136,44 @@ public override CloudEvent DecodeStructuredModeMessage(Stream data, ContentType
         public override CloudEvent DecodeStructuredModeMessage(byte[] data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes) =>
             DecodeStructuredModeMessage(new MemoryStream(data), contentType, extensionAttributes);
 
+        public override async Task<IReadOnlyList<CloudEvent>> DecodeBatchModeMessageAsync(Stream data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes)
+        {
+            data = data ?? throw new ArgumentNullException(nameof(data));
+
+            var jsonReader = CreateJsonReader(data, contentType.GetEncoding());
+            var array = await JArray.LoadAsync(jsonReader).ConfigureAwait(false);
+            return DecodeJArray(array, extensionAttributes);
+        }
+
+        public override IReadOnlyList<CloudEvent> DecodeBatchModeMessage(Stream data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes)
+        {
+            data = data ?? throw new ArgumentNullException(nameof(data));
+
+            var jsonReader = CreateJsonReader(data, contentType.GetEncoding());
+            var array = JArray.Load(jsonReader);
+            return DecodeJArray(array, extensionAttributes);
+        }
+
+        public override IReadOnlyList<CloudEvent> DecodeBatchModeMessage(byte[] data, ContentType contentType, IEnumerable<CloudEventAttribute> extensionAttributes) =>
+            DecodeBatchModeMessage(new MemoryStream(data), contentType, extensionAttributes);
+
+        private IReadOnlyList<CloudEvent> DecodeJArray(JArray jArray, IEnumerable<CloudEventAttribute> extensionAttributes = null)
+        {
+            List<CloudEvent> events = new List<CloudEvent>();
```

Comment:
Should we use jArray.Count when initializing the list?

Reply:
It certainly wouldn't hurt. (Alternatively this method could be rewritten via LINQ... I might have a try at that, as it feels odd to use a loop here.)

Reply:
Done - decided not to use LINQ for the same reasons as in the System.Text.Json code; it seems unnecessarily inefficient and wouldn't actually be much simpler.